routemaster/state_machine/tests/test_state_machine.py

```diff
@@ -160,3 +160,81 @@ def test_stays_in_gate_if_gate_processing_fails(app_config, mock_test_feed):
 
     assert metadata_triggers_processed(app_config, label) is False
     assert current_state(app_config, label) == 'start'
+
+
+def test_concurrent_metadata_update_gate_evaluations_dont_race(create_label, mock_test_feed, app_config, assert_history):
+    test_machine_2 = app_config.config.state_machines['test_machine_2']
+    gate_1 = test_machine_2.states[0]
+
+    label = create_label('foo', 'test_machine_2', {})
+
+    with mock_test_feed():
```

Comment:
I don't think this is needed -- `tests_machine_2` doesn't have any feeds.

Reply:
See also other uses of `mock_test_feed` with `test_machine_2`.

Reply:
6fab74d3cd00583115319da88b9a24248187203e